ektelo/workload.py

```diff
@@ -326,11 +324,6 @@ def __mul__(self, other):
             X, XT = self._Xmatrix(self.weights)
             vect = X.dot(other.weights)
             return MarginalsGram(self.domain, vect)
-        elif isinstance(other, Sum):
```

Comment:
Ok I figured out the root of the performance problem. It's this code here that you removed. When I add it back, example 3 runs quickly again. Is there any reason that it needs to be removed?

Reply:
Good find.  Yes the reason it had to be removed is that it was not quite right.  The details are a bit technical but in this case both self and other should be MarginalsGram objects.  So the issue is that self.domain != other.domain.  Here one is a tuple and the other is a list, resulting in the equality not being satisfied

Reply:
So we can either change the example, or maybe add in checks in the constructor to convert the domain to a tuple